gobblin-data-management/src/test/java/gobblin/data/management/retention/ConfigurableCleanableDatasetTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.retention;
+
+import java.net.URI;
+import java.util.Properties;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import com.google.common.collect.ImmutableMap;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.data.management.retention.dataset.ConfigurableCleanableDataset;
+import gobblin.data.management.retention.policy.NewestKRetentionPolicy;
+import gobblin.data.management.retention.version.DatasetVersion;
+import gobblin.data.management.retention.version.finder.WatermarkDatasetVersionFinder;
+
+public class ConfigurableCleanableDatasetTest {
+
+  @Test
+  public void testConstructor() throws Exception {
+
+    Config conf =
+        ConfigFactory.parseMap(ImmutableMap.<String, String> of("gobblin.retention.version.finder.class",
+            "gobblin.data.management.retention.version.finder.WatermarkDatasetVersionFinder",
+            "gobblin.retention.retention.policy.class",
+            "gobblin.data.management.retention.policy.NewestKRetentionPolicy",
+            "gobblin.retention.newestK.versions.retained", "2"));
```

Comment:
The testing should have multiple versions ( >2 ) and only keep the latest two copies, can not find that testing in the code